src/utilities.jl

```diff
@@ -83,3 +82,24 @@ _ensure_float(::Type{T}) where {T<:AbstractArray} = T
 
 # special case Union{}
 _ensure_float(::Type{Union{}}) = Float64
+
+"""
+$(SIGNATURES)
+
+Check that the *first* argument has all the names required to instantiate a `NamedTuple`
+specified in the second argument.
+
+Currently it only supports `NamedTuple`s for its first argument, and checks there are no
+extra fields. Both requirements may be relaxed in the future.
+
+If the requirements are not met, throw error with an informative message, otherwise
+return `nothing`.
+"""
+function _check_name_compatibility(::Type{<:NamedTuple{A}},
+                                   ::Type{<:NamedTuple{B}}) where {A,B}
+    for b in B
+        b ∈ A || throw(ArgumentError("Property :$(b) not in $(A)."))
```

Comment:
I would suggest making these error messages lazy strings:
```suggestion
        b ∈ A || throw(ArgumentError(LazyString("Property :", b, " not in ", A, ".")))
```